redhat-appstudio/service-provider-integration-operator
# update status of spiaccesscheck when token lookup fails

### What does this PR do?
Reports error to spiaccesscheck status when token lookup fails and don't reconcile CR again.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-161

### How to test this PR?

 1. manually upload the token with test/test credentials for quay.io token
```
curl -v \
  -H 'Content-Type: application/json' \
  -H "Authorization: bearer "$K8S_TOKEN \
  -d "{ \"access_token\": \"test\",\"username\": \"test\"}" \
  $OAUTH_URL'/token/'$SPI_NAMESPACE'/'$SPI_TOKEN_NAME
```
 2. create spiaccesscheck for quay.io repository
 3. you should see error in spiaccesscheck status that lookup failed. If repository is public, it should still be detected.
